### ISSUE TITLE
A service without a defined port generates errors

### ISSUE THREAD

**Issue by dekimsey** [hashicorp/consul-aws#8]
v0.1.1

AWS Cloud Map doesn't seem happy about creating services that have no ports.

Given a minimal service definition of:
```
service {
    name = "foo"
}
```

The consul-aws output is spammed with:
```
2019-02-05T20:19:29.085Z [ERROR] aws: cannot create nodes: error="InvalidInput: A valid port number must be supplied.
```

Setting a port of 0 has no effect.

**Comment by hanshasselberg** [comment]
Good catch! Since cloudmap needs a port, which behaviour would you expect? I think it shouldn't error as much, but apart from that, I am not sure what else to do.

**Comment by dekimsey** [comment]
Honestly I'm not sure. If it takes a 0 port that might work. But I think I set the port to 1 to make the error spam go away. 

I had some unrelated issues with the cloudmap integration and decided to drop CloudMap (and its integration) altogether. (CloudMap response times were slow, unhealthy services were being synced, no mechanism to purge registered services when consul-aws leaves). CloudMap didn't strike me as a ready product. I may revisit this when we get more of our stack on containers.

**Comment by benclark** [comment]
> Since cloudmap needs a port, which behaviour would you expect?

@i0rek, my two cents is the service's sync should just be skipped, silently. We have quite a few services defined without ports, but the ones we actually want in the cloudmap namespace do have ports (and are syncing correctly via consul-aws, fwiw). It's just a log noise issue for us.

**Comment by vanekjar** [comment]
> Since cloudmap needs a port, which behaviour would you expect?

It is not entirely true that Cloud Map requires a port being specified. It all depends on the service [DNS configuration](https://docs.aws.amazon.com/cloud-map/latest/api/API_DnsRecord.html). 

If the service is configured with DNS type `A`, the port is not mandatory. If the service is configured with DNS type `SRV`, both IP and port are mandatory. Please see more details in [Cloud Map documentation](https://docs.aws.amazon.com/cloud-map/latest/api/API_RegisterInstance.html).

I believe the solution is to handle this when registering the instance and provide appropriate attributes (i. e. don't provide port when there is none in Consul).
https://github.com/hashicorp/consul-aws/blob/master/catalog/aws.go#L367
